io-engine/src/grpc/v1/stats.rs

```diff
@@ -114,18 +146,29 @@ impl StatsRpc for StatsService {
                 if let Some(name) = args.name {
                     if let Some(l) = Lvs::lookup(&name) {
                         let stats = l.base_bdev().stats_async().await?;
-                        let io_stat =
-                            IoStats::from(ExternalType((name.clone(), stats)));
+                        let io_stat = IoStats::from(ExternalType((
+                            name.clone(),
+                            l.uuid(),
+                            stats,
+                        )));
                         pool_stats.push(io_stat);
                     }
                 } else {
-                    let bdev_list: Vec<(String, UntypedBdev)> = Lvs::iter()
-                        .map(|lvs| (lvs.name().to_string(), lvs.base_bdev()))
-                        .collect();
-                    for (name, bdev) in bdev_list.iter() {
+                    let bdev_list: Vec<(String, String, UntypedBdev)> =
+                        Lvs::iter()
+                            .map(|lvs| {
+                                (
+                                    lvs.name().to_string(),
+                                    lvs.uuid(),
+                                    lvs.base_bdev(),
+                                )
+                            })
+                            .collect();
+                    for (name, uuid, bdev) in bdev_list.iter() {
                         let bdev_stat = bdev.stats_async().await?;
```

Comment:
We have multiple asyc function in a loop which is serialized. Can we make it parallel using future::join_all() ?

Reply:
Done